pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/multithreading/UnsynchronizedStaticFormatterRule.java

```diff
@@ -48,6 +52,9 @@ public Object visit(ASTFieldDeclaration node, Object data) {
         }
 
         ASTVariableDeclaratorId var = node.getFirstDescendantOfType(ASTVariableDeclaratorId.class);
+        if (var.getType() != null && Arrays.asList(threadSafeFormatter).contains(var.getType().getName())) {
```

Comment:
this check will work on `FastDateFormat`, but not classes that extend `FastDateFormat`, which makes it somewhat brittle.

I'd suggest you use `TypeHelper.isA(TypeNode, String)` directly (you don't even need to check for `var.getType() == null`, the type helper does everything for you.